src/apps/activity_assignments/tests/test_assignments.py

```diff
@@ -0,0 +1,292 @@
+        )
+
+        assert response.status_code == http.HTTPStatus.CREATED, response.json()
+        assignments = response.json()["result"]["assignments"]
+        assert len(assignments) == 1
+        assignment = assignments[0]
+        assert assignment["activityId"] == str(applet_one.activities[0].id)
+        assert assignment["respondentId"] == str(tom.id)
+        assert assignment["targetSubjectId"] == str(lucy_applet_one_subject.id)
+        assert assignment["invitationId"] is None
+        assert assignment["activityFlowId"] is None
+        assert assignment["flowId"] is None
+        assert assignment["id"] is not None
+
+        query = select(ActivityAssigmentSchema).where(ActivityAssigmentSchema.id == assignment["id"])
+        res = await session.execute(query, execution_options={"synchronize_session": False})
+        model = res.scalars().one()
+
+        assert str(model.id) == assignment["id"]
+        assert model.activity_id == applet_one.activities[0].id
+
+    async def test_create_assignment_fail_wrong_activity(
+        self,
+        client: TestClient,
+        applet_one: AppletFull,
+        applet_two: AppletFull,
+        tom: User,
+        session: AsyncSession,
+    ):
+        client.login(tom)
+
+        assignments_create = ActivitiesAssignmentsCreate(
+            assignments=[
+                ActivityAssignmentCreate(activity_id=applet_two.activities[0].id, respondent_id=tom.id)
+            ]  # activity from applet two
+        )
+
+        response = await client.post(
+            self.activities_assignments_applet.format(applet_id=applet_one.id), data=assignments_create
+        )
+
+        assert response.status_code == http.HTTPStatus.BAD_REQUEST, response.json()
+        result = response.json()["result"][0]
+        assert result["message"] == f"Invalid activity id {applet_two.activities[0].id}"
+
+    async def test_create_multiple_assignments_with_flow(
+        self,
+        client: TestClient,
+        applet_one_with_flow: AppletFull,
+        tom: User,
+        lucy_applet_one_subject: SubjectFull,
+        session: AsyncSession,
+    ):
+        client.login(tom)
+
+        assignments_create = ActivitiesAssignmentsCreate(
+            assignments=[
+                ActivityAssignmentCreate(activity_id=applet_one_with_flow.activities[0].id, respondent_id=tom.id),
+                ActivityAssignmentCreate(
+                    activity_flow_id=applet_one_with_flow.activity_flows[0].id,
+                    respondent_id=tom.id,
+                    target_subject_id=lucy_applet_one_subject.id,
+                ),
+            ]
+        )
+
+        response = await client.post(
+            self.activities_assignments_applet.format(applet_id=applet_one_with_flow.id), data=assignments_create
+        )
+
+        assert response.status_code == http.HTTPStatus.CREATED, response.json()
+        assignments = response.json()["result"]["assignments"]
+        assert len(assignments) == 2
```

Comment:
Just thought it prudent to include these checks, just to make sure these are working properly:
- omitting target subject
- assignment for activity flows

```suggestion
        assert len(assignments) == 2
        
        assignment = assignments[0]
        assert assignment["activityId"] == str(applet_one_with_flow.activities[0].id)
        assert assignment["respondentId"] == str(tom.id)
        assert assignment["targetSubjectId"] is None
        assert assignment["invitationId"] is None
        assert assignment["activityFlowId"] is None

        assignment = assignments[1]
        assert assignment["activityId"] is None
        assert assignment["respondentId"] == str(tom.id)
        assert assignment["targetSubjectId"] == str(lucy_applet_one_subject.id)
        assert assignment["invitationId"] is None
        assert assignment["activityFlowId"] == str(applet_one_with_flow.activity_flows[0].id)
```